### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by lthibault**
Does what it says on the tin 🙃

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
-          GOOS: linux
-          GOARCH: 386
+          go-version: ${{ matrix.go-version }}
+          # You can test your matrix by printing the current Go version
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+      # - name: Lint contracts
+      #   run: cd suave && forge fmt --check
+      # - name: Build contracts
+      #   run: cd suave && forge build
```
Comment (sukoneck) [inline .github/workflows/go.yml:34]: do you still need these?

```suggestion
```

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -1,23 +1,43 @@
...
+      - name: Display Go version
+        run: go version
+      - name: Run unit tests
+        run: go test -short ./accounts ./cmd/geth ./core ./core/types ./core/vm ./eth/... ./internal/ethapi/... ./miner ./params #./suave/...
```
Comment (sukoneck) [inline .github/workflows/go.yml:38]: do you still need this?
```suggestion
        run: go test -short ./accounts ./cmd/geth ./core ./core/types ./core/vm ./eth/... ./internal/ethapi/... ./miner ./params
```

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -1,23 +1,43 @@
...
+  CGO_CFLAGS: "-O -D__BLST_PORTABLE__"
+
 jobs:
   build:
-    runs-on: self-hosted
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [ '1.20', '1.21.x' ]
+        goarch: ["386", amd64, arm64]
```
Comment (sukoneck) [inline .github/workflows/go.yml:20]: note- this is going to do 6 runs each for each PR, so let's keep an eye on performance.

**Review by sukoneck — APPROVED** [review]

**Comment by lthibault** [comment]
Okay, I think the remaining failures are spurious.  I mucked with this a fair bit since @sukoneck last looked.  Can I get one more review?

**Review by sukoneck — APPROVED** [review]
🔥

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -1,23 +1,29 @@
...
-    branches: [ master ]
+    branches: [ main ]
   workflow_dispatch:
 
+env:
+  CGO_CFLAGS_ALLOW: "-O -D__BLST_PORTABLE__"
+  CGO_CFLAGS: "-O -D__BLST_PORTABLE__"
+
 jobs:
   build:
```
Comment (metachris) [inline .github/workflows/go.yml:15]:
```suggestion
  test:
```

**Comment by metachris** [comment]
that's nice! 

left two minor comments/suggestions to make it more comprehensible and better triggered, see also https://github.com/flashbots/suave-geth/blob/main/.github/workflows/checks.yml and https://github.com/flashbots/go-template/blob/main/.github/workflows/checks.yml for our general patterns.

I'd almost recommend to rename `go.yaml` into `checks.yaml` like we have in the other repos, so that the filename already is somewhat descriptive in itself.

**Comment by lthibault** [comment]
@metachris Thanks for the suggestions.  Regarding the number of tests, I've gone ahead and removed the `goarch` parameter because I think we're more sensitive to incompatibilities with Go versions than target architectures.  Tests pass pretty quickly now.

Merging.

**Pull request merged**